src/server/download.ts

```diff
@@ -21,14 +21,16 @@ import { Artifact } from './artifact';
 
 export class Download {
   readonly artifact: Artifact;
+  readonly uuid: string;
   readonly url: string;
   private _page: Page;
   private _suggestedFilename: string | undefined;
 
   constructor(page: Page, downloadsPath: string, uuid: string, url: string, suggestedFilename?: string) {
     const unaccessibleErrorMessage = !page._browserContext._options.acceptDownloads ? 'Pass { acceptDownloads: true } when you are creating your browser context.' : undefined;
-    this.artifact = new Artifact(page, path.join(downloadsPath, uuid), unaccessibleErrorMessage);
+    this.artifact = new Artifact(page, path.join(downloadsPath, uuid), unaccessibleErrorMessage, this);
     this._page = page;
+    this.uuid = uuid;
```

Comment:
Once you change the artifact to accept `CancelCallback` function this field should not be necessary.